src/hetzner/instance/create.cr

```diff
@@ -294,14 +294,20 @@ class Hetzner::Instance::Create
     commands
   end
 
-  def self.generate_post_create_commands_str(settings, snapshot_os, additional_post_create_commands, init_commands)
+  def self.generate_post_create_commands_str(settings, snapshot_os, additional_post_create_commands, init_commands, for_cluster_autoscaler = false)
     post_create_commands = mandatory_post_create_commands(settings).dup
 
     add_microos_commands(post_create_commands) if snapshot_os == "microos"
 
     formatted_additional_commands = format_additional_commands(additional_post_create_commands)
 
-    combined_commands = [post_create_commands, init_commands, formatted_additional_commands].flatten
+    # For cluster autoscaler (HCLOUD_CLOUD_INIT), run additional commands first to ensure network setup
+    # happens before other initialization commands
+    combined_commands = if for_cluster_autoscaler
+                          [formatted_additional_commands, post_create_commands, init_commands].flatten
+                        else
+                          [post_create_commands, init_commands, formatted_additional_commands].flatten
+                        end
```

Comment:
Have you tested different scenarios to make sure this change won’t mess things up when the public interface is around or in other cases?